### DISCUSSION TITLE
Post-Edit Command Hook

### DISCUSSION THREAD

**Discussion by hugginsio** [zk-org/zk#366] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

A useful add-on to my productivity workflow would be the ability for `zk` to execute scripts before and after commands. Really, I'd like to run a script after a note is edited.

You can sort of do this with aliases that shadow existing commands:

```toml
[alias]
edit = 'zk edit "$@" && zk-edit-hook'
```

But I'd really like to pass a list of edited files to the post-command hook, perhaps by an environment variable.

Would you entertain this functionality being added to `zk`?

**Comment by tjex** [comment]
I would imagine that this would be better suited as feature for your editor no? eg, in neovim a `BufWrite` autocommand?

**Reply by hugginsio** [reply]
That is a potential solution here - and what I will probably do in the meantime - but with zk's focus on being automation-friendly and editor agnostic, it felt like the CLI was a natural place for this functionality to reside.

Extending neovim to do this isn't much of a challenge, but the same might not be true in other editors.
